modules/quantcastIdSystem.js

```diff
@@ -7,11 +7,156 @@
 
 import {submodule} from '../src/hook.js'
 import { getStorageManager } from '../src/storageManager.js';
+import { triggerPixel, logInfo } from '../src/utils.js';
+import { uspDataHandler, coppaDataHandler, gdprDataHandler } from '../src/adapterManager.js';
 
 const QUANTCAST_FPA = '__qca';
+const DEFAULT_COOKIE_EXP_TIME = 392; // (13 months - 2 days)
+const PREBID_PCODE = 'p-KceJUEvXN48CE'; // Not associated with a real account
+const DOMAIN_QSERVE = 'https://pixel.quantserve.com/pixel';
+const QUANTCAST_VENDOR_ID = '11';
+const PURPOSE_DATA_COLLECT = '1';
+const PURPOSE_PRODUCT_IMPROVEMENT = '10';
+
+var clientId;
+var cookieExpTime;
 
 export const storage = getStorageManager();
 
+export function firePixel() {
+  // check for presence of Quantcast Measure tag _qevent obj
+  if (!window._qevents) {
+    const gdprPrivacyString = gdprDataHandler.getConsentData();
+    const usPrivacyString = uspDataHandler.getConsentData();
+
+    var fpa = storage.getCookie(QUANTCAST_FPA);
+    var fpan = '0';
+    var now = new Date();
+    var domain = quantcastIdSubmodule.findRootDomain();
+    var et = now.getTime();
+    var tzo = now.getTimezoneOffset();
+    var usPrivacyParamString = '';
+    var firstPartyParamStrings;
+    var gdprParamStrings;
+
+    if (!(hasGDPRConsent(gdprPrivacyString) && hasCCPAConsent(usPrivacyString))) {
+      var expired = new Date(0).toUTCString();
+      fpan = 'u';
+      fpa = '';
+      storage.setCookie(QUANTCAST_FPA, fpa, expired, '/', domain, null);
+    } else if (!fpa) {
+      var expires = new Date(now.getTime() + (cookieExpTime * 86400000)).toGMTString();
```

Comment:
somewhere we need to be checking we have a valid clientId before we set fpa or call quantserve

Reply:
(please add tests to cover this)

Reply:
So this isn't your fault and this is in suit with the current status-quo in Prebid.js, but this is a little maddening.

```
> git grep '* 24'
modules/apstreamBidAdapter.js:    var daysSinceApEpoch = Math.floor(timeDiff / (1000 * 3600 * 24));
modules/categoryTranslation.js:    if (!mappingData || timestamp() > mappingData.lastUpdated + refreshInDays * 24 * 60 * 60 * 1000) {
modules/criteoIdSystem.js:const cookiesMaxAge = 13 * 30 * 24 * 60 * 60 * 1000;
modules/id5IdSystem.js:  return (new Date(Date.now() + (1000 * 60 * 60 * 24 * expDays))).toUTCString();
modules/lotamePanoramaIdSystem.js:const DAY_MS = 60 * 60 * 24 * 1000;
modules/parrableIdSystem.js:const ONE_YEAR_MS = 364 * 24 * 60 * 60 * 1000;
modules/pubCommonIdSystem.js:      const expiresStr = (new Date(Date.now() + (storage.expires * (60 * 60 * 24 * 1000)))).toUTCString();
modules/userId/index.js:    const expiresStr = (new Date(Date.now() + (storage.expires * (60 * 60 * 24 * 1000)))).toUTCString();
modules/userId/index.js:    const expiresStr = (new Date(Date.now() + (CONSENT_DATA_COOKIE_STORAGE_CONFIG.expires * (60 * 60 * 24 * 1000)))).toUTCString();
```

Reply:
@ssmccoy - Yeah, this could have been one common const somewhere. 

Btw, Any objections to also change 86400000 to 60 * 60 * 24 * 1000? I think it is at least a little clearer.